readthedocs/proxito/middleware.py

```diff
@@ -22,6 +22,59 @@
 log = structlog.get_logger(__name__)  # noqa
 
 
+def _get_domain_from_host(host):
+    """
+    Get the normalized domain from a hostname.
+
+    A hostname can include the port.
+    """
+    return host.lower().split(":")[0]
+
+
+def _unresolve_domain(domain):
+    """
+    Unresolve domain.
+
+    :param str domain: Domain to extrac the project slug from.
+    :returns: A tuple with the project slug, domain object, and if the domain
+     is external.
+    """
+    public_domain = _get_domain_from_host(settings.PUBLIC_DOMAIN)
+    external_domain = _get_domain_from_host(settings.RTD_EXTERNAL_VERSION_DOMAIN)
+
+    subdomain, *rest_of_domain = domain.split(".", maxsplit=1)
+    rest_of_domain = rest_of_domain[0] if rest_of_domain else ""
+
+    if public_domain in domain:
+        # Serve from the PUBLIC_DOMAIN, ensuring it looks like `foo.PUBLIC_DOMAIN`.
+        if public_domain == rest_of_domain:
+            project_slug = subdomain
+            return project_slug, None, False
+
+        # TODO: This can catch some possibly valid domains (docs.readthedocs.io.com) for example,
+        # but these might be phishing, so let's ignore them for now.
```

Comment:
It would be good to log these for now.

Reply:
This is logged at https://github.com/readthedocs/readthedocs.org/blob/185771d2fabcdefa6519698d1e43e82e653bcad6/readthedocs/proxito/middleware.py#L110-L110, I wasn't sure if to log this from the inside or outside the function.